File: api/F1CompanionApi/Data/Entities/Race.cs
using Microsoft.EntityFrameworkCore;

namespace F1CompanionApi.Data.Entities;

[Index(nameof(SeasonId), nameof(Round), IsUnique = true)]
public class Race : BaseEntity
{
    public required int SeasonId { get; set; }
    public required int Round { get; set; }
    public required string Name { get; set; }
    public required string Location { get; set; }
    public required string Circuit { get; set; }
    public required string Country { get; set; }
    public required DateTime RaceDate { get; set; }
    public DateTime? LockDeadline { get; set; }

    public Season Season { get; set; } = null!;
}

File: web/src/contracts/Race.ts
import type { Circuit } from './Circuit';

export type WeekendFormat = 0 | 1;

export interface Race {
  id: number;
  seasonId: number;
  round: number;
  name: string;
  circuit: Circuit;
  raceDate: string;
  lockDeadline: string | null;
  isCurrent: boolean;
  weekendFormat: WeekendFormat;
}
